### PR TITLE
Replacing Zombie JS with Puppeteer, remove mocha in favour of jest

### PR DESCRIPTION
**Pull request by tomrosier**
What
----

- Replace Zombie.js with Puppeteer
- Replace Mocha with Jest to be consistence with the rest of the test suite.
- Update readme, to explain how to run the acceptance tests locally against the dev environments.

How to review
-------------

To run the tests follow the section "Executing the acceptance tests against dev environment:" in the new read me.

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

**Review by AP-Hunt — APPROVED** [review]
Paired with Tom, got the problematic tests passing. The change itself is good.

File: README.md [hunk 1]
```diff
@@ -71,6 +71,21 @@ Execute the unit tests to ensure everything looks good:
 npm test
 ```
 
+Executing the acceptance tests against dev environment:
```
Comment (kr8n3r) [inline README.md:74]: thanks for this addition

File: acceptance-test-jest-config.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (kr8n3r) [inline acceptance-test-jest-config.json:1]: i think you can have much smaller config, as it only needs a config for one type of test with a specific setup.
still, it seems to work

File: package.json [hunk 3]
```diff
@@ -55,6 +56,7 @@
     "@types/passport": "^1.0.7",
     "@types/passport-oauth2": "^1.4.11",
     "@types/pino": "^6.3.9",
+    "@types/puppeteer": "^5.4.4",
```
Comment (kr8n3r) [inline package.json:59]: since v7 Puppeteer ship their own types https://github.com/puppeteer/puppeteer#usage-with-typescript

File: package.json [hunk 4]
```diff
@@ -80,29 +82,29 @@
...
     "lodash": "^4.17.21",
     "mini-css-extract-plugin": "^2.1.0",
-    "mocha": "^8.4.0",
     "nock": "^13.1.1",
     "nodemon-webpack-plugin": "^4.5.2",
     "pegjs": "^0.10.0",
     "pegjs-loader": "^0.5.6",
     "pino-pretty": "^5.1.1",
+    "puppeteer": "^9.1.1",
+    "request-promise-native": "^1.0.9",
```
Comment (kr8n3r) [inline package.json:95]: this is a deprecated package, we're hoping to remove. I assume it was added to resolve failing [app.test](https://github.com/alphagov/paas-admin/blob/main/src/components/app/app.test.ts#L5)

removing that line would require a rewrite of those tests

**Review by kr8n3r — COMMENTED** [review]
while I have some comments, we've had this work going on for a while and I might go on longer so let's just get it done and look to addressing things as part of the PaaS squad

**Review by 0atman — APPROVED** [review]
Looks great, agreed that @kr8n3r 's changes could be chalked up to debt and handled in a subsequent PR. I'll capture those as issues :ok_hand:

File: acceptance-test-jest-config.json (hunk 2 above)
Comment (paroxp) [reply acceptance-test-jest-config.json:1]: Addressing

File: package.json (hunk 3 above)
Comment (paroxp) [reply package.json:59]: Addressing

File: package.json (hunk 4 above)
Comment (paroxp) [reply package.json:95]: I removed the dependency. The refectory of `app.test` should be addressed outside of this story.

File: .eslintrc [hunk 5]
```diff
@@ -49,7 +49,7 @@
     "@typescript-eslint/ban-ts-comment": "warn",
     "@typescript-eslint/explicit-function-return-type": "warn",
     "@typescript-eslint/naming-convention": ["warn", {"selector": "interface", "format": ["PascalCase"], "custom": {"regex": "^I[A-Z]", "match": true}}],
-    "@typescript-eslint/no-floating-promises": "error",
+    "@typescript-eslint/no-floating-promises": "off",
```
[the remark this answers is not in the document]
Comment (paroxp) [reply .eslintrc:52]: Addressing

[the remark this answers is not in the document]
Comment (0atman) [reply .eslintrc]: you're doing gods work :ok_hand:

**Pull request merged**